common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -597,6 +597,8 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
     HADOOPNUMREDUCERS("mapreduce.job.reduces", -1, "", true),
 
     // Metastore stuff. Be sure to update HiveConf.metaVars when you add something here!
+    METASTOREDBTYPE("hive.metastore.db.type", "DERBY", new StringSet("DERBY", "ORACLE", "MYSQL", "MSSQL", "POSTGRES"),
+        "Type of database used for the metastore"),
```

Comment:
I'll create a follow up for it. It's not as trivial as it should be, as far as I remember. The code we have needs a connection and is running queries to find out. The StorageHandler was written in a way that needs to know the product first and will then create a connection.